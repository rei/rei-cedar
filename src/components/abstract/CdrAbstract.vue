<script setup lang="ts">
import { useCssModule } from 'vue';

defineOptions({
  name: 'CdrAbstract',
});

defineProps({
    /** Sets the HTML tag (p, span, h1, h2, etc) */
    tag: {
      type: String,
      default: 'p',
    },
});

const baseClass = 'cdr-abstract';
const style = useCssModule();
</script>

<template>
  <component
    :is="tag"
    :class="style[baseClass]"
  >
    <slot />
  </component>
</template>

<style lang="scss" module src="./styles/CdrAbstract.module.scss">
</style>
