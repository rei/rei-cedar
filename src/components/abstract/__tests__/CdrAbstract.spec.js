import { shallowMount } from '../../../../test/vue-jest-style-workaround.js';
import CdrAbstract from 'componentdir/abstract/CdrAbstract';

describe('CdrAbstract', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(CdrAbstract);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('has a failing test by default so you remember to do them', () => {
    expect(false).toBe(true);
  });
});
