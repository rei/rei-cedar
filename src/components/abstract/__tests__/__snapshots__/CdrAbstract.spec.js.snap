// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrAbstract > matches snapshot 1`] = `
<p
  class="cdr-abstract"
>
  
  This is an abstract
  
</p>
`;
